bird/handlers/main.yml

```diff
@@ -1,7 +1,27 @@
 ---
-- name: configure bird
-  shell: birdc configure || systemctl restart bird
+- name: reload bird
```

Comment:
Reload passt nicht. Angenommen bird lief vorher nicht, dann tut reload nichts.

Reply:
Bird selbst die Neukonfiguration einlesen zu lassen, sorgt dafür, dass der Übergang weich ist und keine Verbindungen abreißen. Ich weiß nicht genau, wie reload sich da verhält.

Reply:
Wenn ich es richtig sehe, dann geht ihr bei allen Rollen davon aus, dass der jeweilige Dienst bereits läuft und so konfiguriert ist, dass er beim Systemstart mit gestartet wird. Das ist die Debian/Ubuntu-Standardeinstellung für frisch installierte Dienste.
Wenn bird hier nicht läuft, dann wird er auch nur neu gestartet wenn die Konfiguration geändert wurde.

Ein "systemctl reload" soll stets den Dienst nicht unterbrechen sondern nur neue Konfiguration einlesen. So auch hier: "systemctl reload bird" führt "birdc configure" aus (siehe /lib/systemd/system/bird.service, Zeile ExecReload). Im Prinzip ist es also egal. Die Ansible-Doku empfiehlt halt, dass man möglichst die Module (hier das systemd-Modul) nutzt und möglichst wenig Shell-Befehle.

Reply:
OK, thanks for the details.